api/src/main/java/org/apache/iceberg/expressions/ExpressionUtil.java

```diff
@@ -66,6 +67,40 @@ public static String toSanitizedString(Expression expr) {
     return ExpressionVisitors.visit(expr, StringSanitizer.INSTANCE);
   }
 
+  /**
+   * Returns whether two unbound expressions will accept the same inputs.
+   * <p>
+   * If this returns true, the expressions are guaranteed to return the same evaluation for the same input. However, if
+   * this returns false the expressions may return the same evaluation for the same input. That is, expressions may
+   * be equivalent even if this returns false.
+   *
+   * @param left an unbound expression
+   * @param right an unbound expression
+   * @param struct a struct type for binding
+   * @return true if the expressions are equivalent
+   */
+  public static boolean equivalent(Expression left, Expression right, Types.StructType struct) {
```

Comment:
Do we care about case sensitivity in any of the methods?

Reply:
Maybe? My thought was that since this is binding both expressions, the important thing is that we use the setting consistently. But you're right that we should probably just pass it in because the expressions may not match the struct.

Reply:
Added.